### PR TITLE
Async-related changes, mostly to foster communications between a (full python3) host and one or more Micropython-based microcontrollers.

### PR DESCRIPTION
**Pull request by bapowell**
- Changed uasyncio to asyncio.
  - Provided a new example, asyntest_py3_serial.py, to demonstrate running under full python3.
- Changed as_load.py.
  - Added missing _file() function.
  - Added 'observer' option for aload.
- Added as_loader.py.
  - New aloader class, which offers the same functionality as aload, but in a cleaner, objected-oriented manner.

### CODE REVIEW

**Comment by peterhinch** [comment]
Thanks for that, it looks like a useful enhancement. I will study this later in the week.

**Comment by peterhinch** [comment]
Re running under CPython. Please could you share the motivation for this. For context please see https://github.com/peterhinch/micropython-msgpack/issues/3#issuecomment-1506506727 where I outline my reasons for not going down this route.

**Comment by bapowell** [comment]
Yours supports asyncio streams (aload), and it works with no change whatsoever between CPython and Micropython. So I'm able to use the same library for both code bases: host vs microcontroller, with serial comms between them.

File: asyntest.py
Comment (peterhinch) [inline asyntest.py:1]: A better way to detect the platform is
```py
from sys import platform
if platform == "pyboard":
    uart = UART(4, 9600)
elif platform == "rp2:
    uart = UART(0, baudrate=9600, tx=Pin(0), rx=Pin(1))  # Pi Pico (link pins 0 and 1)
else:
    raise OSError(f"Unknown platform {platform}") 
```
Or other exception if you think OSError is wrong. This detection method is more robust and easily extensible e.g. to ESP32.

**Review by peterhinch — CHANGES_REQUESTED** [review]
Aside from my comment on platform detection I like this. Nice work.

File: asyntest.py
Comment (bapowell) [reply asyntest.py:1]: I agree that's a much better way.
I just pushed a commit, per your suggestion. I also included esp32 (and tested it) while I was at it.

**Comment by peterhinch** [comment]
Thank you.

**Comment by peterhinch** [comment]
Thanks also for pointing out the bug :)

I've pushed an update with the following changes:
1. The package may be installed using `mpremote`.
2. `README.md` explains installation options and lists files with their purpose.
3. Bug fixed. My editor is configured to run Black on changed files, so there is PEP8 formatting.

**Comment by peterhinch** [comment]
@bapowell In case you're still following this project, I'm planning a significant upgrade. I've pushed my current work to a branch `modular-extnsions`.

The objectives are in the docs, but some of the major changes are:
* Simplify creation of extensions. Each extension of a built-in now has an individual very simple module.
* Remove the original async code in favour of your `ALoader`.
* Asynchronous decode now uses an asynchronous iterator.
* The `Observer` object can now be any callable and its incoming data includes a delimiter.

Sadly it's a breaking change for async users and creators of extensions.

Comments welcome.

**Comment by bapowell** [comment]
Those all sound like good changes, Peter.

I skimmed the commits in your `modular-extensions` branch, and don't see anything that's confusing or concerning.

Breaking changes, yes, but shouldn't be terribly difficult for users to refactor their code accordingly.

**Comment by peterhinch** [comment]
Thanks for the feedback. I'm still doing a few minor code simplifications but I will release soon.

**Pull request merged**